controllers/dscinitialization/servicemesh_setup.go

```diff
@@ -12,37 +13,42 @@ import (
 )
 
 func (r *DSCInitializationReconciler) configureServiceMesh(instance *dsciv1.DSCInitialization) error {
-	switch instance.Spec.ServiceMesh.ManagementState {
-	case operatorv1.Managed:
-		serviceMeshFeatures := feature.ClusterFeaturesHandler(instance, configureServiceMeshFeatures())
-
-		if err := serviceMeshFeatures.Apply(); err != nil {
-			r.Log.Error(err, "failed applying service mesh resources")
-			r.Recorder.Eventf(instance, corev1.EventTypeWarning, "DSCInitializationReconcileError", "failed applying service mesh resources")
-			return err
-		}
-	case operatorv1.Unmanaged:
-		r.Log.Info("ServiceMesh CR is not configured by the operator, we won't do anything")
-	case operatorv1.Removed:
-		r.Log.Info("existing ServiceMesh CR (owned by operator) will be removed")
-		if err := r.removeServiceMesh(instance); err != nil {
-			return err
+	if instance.Spec.ServiceMesh != nil {
```

Comment:
So we are keeping the nested `if`? I am confused now as there's also this comment https://github.com/opendatahub-io/opendatahub-operator/pull/885/files#r1535380592 :)

Reply:
looks like i did not check the changes after our conversation :/

but
in order to print this` "ServiceMesh is not configed in DSCI, same as default to 'Removed',"  `will need a `nil` check, or ? :thinking: